NMCA_001/NMCA_001_script.lua

```diff
@@ -0,0 +1,767 @@
+        ScenarioFramework.GiveUnitToArmy(ScenarioInfo.Engineer3, Player3)
+        ScenarioFramework.GiveUnitToArmy(ScenarioInfo.Engineer4, Player4)
+    end
+
+	WaitSeconds(1)
+	ForkThread(M1)
+end
+
+
+function M1()
+	ScenarioFramework.CreateTimerTrigger(M1StartUEFScouting, UEFM1Timer[Difficulty])
+
+	local function MissionNameAnnouncement()
+        ScenarioFramework.SimAnnouncement(ScenarioInfo.name, 'mission by [e]JJs_AI')
+    end
+
+    ScenarioFramework.CreateTimerTrigger(MissionNameAnnouncement, 7)
+
+    ScenarioInfo.M1P1 = Objectives.Protect(
+        'primary',                    -- type
+        'incomplete',                   -- complete
+        'Defend Command Ship',  -- title
+        'Defend the Command Ship at all costs. We need it to leave this planet.',  -- description
+        {                               -- target
+            Units = {ScenarioInfo.Ship},
+        }
+    )
+    table.insert(AssignedObjectives, ScenarioInfo.M1P1)
+
+	ScenarioFramework.CreateArmyIntelTrigger(ForkAttackThread, ArmyBrains[UEF], 'LOSNow', false, true,  categories.NOMADS, true, ArmyBrains[Player1] )
+end
+
+function M1StartUEFScouting()
+	----------
+	-- Scouting Starts
+	----------
+	local Scouts = ScenarioUtils.CreateArmyGroupAsPlatoon('UEF', 'Scouts','NoFormation')
+
+	for k, v in Scouts:GetPlatoonUnits() do
+    	ScenarioFramework.GroupPatrolRoute({v}, ScenarioPlatoonAI.GetRandomPatrolRoute(ScenarioUtils.ChainToPositions('M1_UEF_Scout_Chain')))
+    end
+end
+
+function ForkAttackThread()
+	ForkThread(M1StartUEFAttacks)
+end
+
+function M1StartUEFAttacks()
+	if not Spotted then
+		ScenarioFramework.Dialogue(OpStrings.UEF_Notice, nil, true)
+		
+		WaitSeconds(7)	
+
+		local VisMarker1_1 = ScenarioFramework.CreateVisibleAreaLocation(10, 'M1_Vis_1_1', 0, ArmyBrains[Player1])
+		ScenarioInfo.TacLauncher = ScenarioInfo.UnitNames[UEF]['TacMissile']
+
+		Cinematics.EnterNISMode()
+
+		Cinematics.CameraMoveToMarker(ScenarioUtils.GetMarker('Cutscene_01c'), 0)
+		WaitSeconds(4)
+		IssueTactical({ScenarioInfo.TacLauncher}, ScenarioInfo.Ship)
+		WaitSeconds(5)	
+
+		Cinematics.CameraMoveToMarker(ScenarioUtils.GetMarker('Cutscene_01b'), 2)
+
+		VisMarker1_1:Destroy()
+
+		Cinematics.ExitNISMode()
+
+		WaitSeconds(10)	
+		ScenarioFramework.Dialogue(OpStrings.M1_Scream_Incoming, nil, true)
+		WaitSeconds(9)
+		ScenarioFramework.Dialogue(OpStrings.M1_Update_Commander, nil, true)
+
+	    Spotted = true
+
+		----------
+		-- Start Attacks
+		----------
+		M1OutpostAI.M1LandAttacks()
+		M1OutpostAI.M1AirAttacks()
+
+		----------
+		-- Set Timer for Missile Launches
+		----------
+		ScenarioFramework.CreateUnitDeathTrigger(TacDead, ScenarioInfo.TacLauncher)
+		ScenarioFramework.CreateTimerTrigger(LaunchTac, TMLTimer[Difficulty])
+		ScenarioFramework.CreateTimerTrigger(M1Reinforcements, M1TransportTimer[Difficulty])
+
+		WaitSeconds(10)
+
+        UEFM1Taunts()
+
+		ForkThread(M2)
+
+        WaitSeconds(7)
+        ScenarioFramework.PlayUnlockDialogue()

```

Comment:
use `ScenarioFramework.RemoveRestrictionForAllHumans(categories, VOflag)`
if VOflag is true it will plaz the unlock dialogue